core-web/libs/sdk/experiments/src/lib/utils/utils.ts

```diff
@@ -123,15 +122,85 @@ export const checkFlagExperimentAlreadyChecked = (): boolean => {
  * Checks if the data needs to be invalidated based on the creation date.
  *
  * @returns {boolean} - True if the data needs to be invalidated, false otherwise.
- * @param dbData
  */
-export const checkInvalidateDataChecked = (dbData: IndexDbStoredData): boolean => {
-    if (!dbData) {
+export const isDataCreateValid = (): boolean => {
+    try {
+        const timeValidUntil = Number(localStorage.getItem(EXPERIMENT_FETCH_EXPIRE_TIME_KEY));
+        if (isNaN(timeValidUntil)) {
+            return false;
+        }
+
+        const now = Date.now();
+
+        return timeValidUntil > now;
+    } catch (error) {
+        console.error('Error checking data validity', error);
+
         return false;
     }
+};
 
-    const { created } = dbData;
-    const now = Date.now();
+/**
+ * Ad to an absolute path the baseUrl depending on the location.
+ *
+ * @param {string | null} absolutePath - The absolute path of the URL.
+ * @param {Location} location - The location object representing the current URL.
+ * @returns {string | null} - The full URL or null if absolutePath is null.
+ */
+export const getFullUrl = (location: Location, absolutePath: string | null): string | null => {
+    if (absolutePath === null) {
+        return null;
+    }
+
+    if (!isFullUrl(absolutePath)) {
+        const baseUrl = location.origin;
+
+        return `${baseUrl}${absolutePath}`;
+    }
+
+    return absolutePath;
+};
+
+const isFullUrl = (url: string): boolean => {
+    const pattern = /^https?:\/\//i;
+
+    return pattern.test(url);
+};
+
+/**
+ * Updates the URL with the queryParam with the experiment variant name.
+ *
+ * @param {Location|string} location - The current location object or the URL string.
+ * @param {Variant} variant - The experiment variant to update the URL with.
+ * @returns {string} The updated URL string.
+ */
+export const updateUrlWithExperimentVariant = (
+    location: Location | string,
+    variant: Variant
+): string => {
+    const href = typeof location === 'string' ? location : location.href;
+    const url = new URL(href);
+
+    if (variant !== null) {
+        const params = url.searchParams;
+        params.set(EXPERIMENT_QUERY_PARAM_KEY, variant.name);
+        url.search = params.toString();
+    }
+
+    return url.toString();
+};
+
+/**
+ * Check if two arrays of Experiment objects are equal.
+ *
+ * @param {Experiment[]} obj1 - The first array of Experiment objects.
+ * @param {Experiment[]} obj2 - The second array of Experiment objects.
+ * @return {boolean} - True if the arrays are equal, false otherwise.
+ */
+export const objectsAreEqual = (obj1: Experiment[], obj2: Experiment[]): boolean => {
+    if (obj1.length === 0 && obj2.length === 0) {
+        return false;
+    }
 
-    return now - created > LOCAL_STORAGE_TIME_DURATION_MILLISECONDS;
+    return JSON.stringify(obj1) === JSON.stringify(obj2);
```

Comment:
Is this strong enough, @oidacra? Is the properties are in a different order, this will return `false`

Reply:
I don't think it's the strongest, but it's the one that gives me the best performance. I didn't want to use an external library, I don't know if there is any method in the repo that allows me to compare objects like that.

Reply:
After a conversation, I will leave it like this, if triggers a lot of false positives, we will take a look at a better way.

Reply:
We can keep the current implementation for now. However, if we encounter a significant number of `false` negatives, we should consider using a third-party library like `lodash` to compare objects